### ISSUE TITLE
Log Item Updates in Targets

### ISSUE THREAD

**Issue by KirillOsenkov** [Microsoft/msbuild#3538]
See the bug report here:
https://github.com/KirillOsenkov/MSBuildStructuredLog/issues/182

I'm guessing we should be logging when a target updates an item, and we don't see anything logged.

**Comment by dagood** [comment]
Does this include when metadata is updated on all items without an `Update` attribute? Like this:

```xml
<A Include="A.txt;B.cs;C.log" />
<A Other="common%(Extension)" />
```

I assume no `Update` is basically an implicit `Update="@(A)"`.

This is a pain point in debugging for me... hard to keep track of where updates come from. (If a later `Include` uses the items, it prints the metadata, and I can add a throwaway item list intentionally to debug stuff. But when I don't own the code and I'm not sure which of many targets might be updating something, it can be very hard to track down.) /cc @rainersigwald

**Comment by rainersigwald** [comment]
Yes; `Update` is always completely ignored inside a target (long sad story, see #2835).

**Comment by KirillOsenkov** [comment]
If or when we log items, we should use TaskParameterEventsArgs with a new TaskParameterMessageKind.UpdateItem enum field.

Alternatively, there's a cheaper suggestion to just log a message and not log all items:
https://github.com/dotnet/msbuild/issues/12492#issuecomment-3259237400
